core/src/test/scala/integration/kafka/api/PlaintextConsumerTest.scala

```diff
@@ -187,11 +187,15 @@ class PlaintextConsumerTest extends BaseConsumerTest {
     assertEquals(1, listener.callsToAssigned)
     assertEquals(0, listener.callsToRevoked)
 
-    Thread.sleep(3500)
+    // as long as we are part of max.poll no rebalance should be triggered
```

Comment:
I looked at the flakiness of this test, and propose we just check for exact num.callback